### ISSUE TITLE
Build dependency happy fails on OSX 10.15.7

### ISSUE THREAD

**Issue by capital-G** [kowainik/learn4haskell#373]
See https://github.com/simonmar/happy/issues/180 and 

```
➜ make test-chapter2
cabal test doctest-chapter2 --enable-tests --test-show-details=direct
Build profile: -w ghc-8.10.1 -O1
In order, the following will be built (use -v for more details):
 - learn4haskell-0.0.0.0 (test:doctest-chapter2) (first run)
Preprocessing test suite 'doctest-chapter2' for learn4haskell-0.0.0.0..
Building test suite 'doctest-chapter2' for learn4haskell-0.0.0.0..
Running 1 test suites...
Test suite doctest-chapter2: RUNNING...
Examples: 54  Tried: 54  Errors: 0  Failures: 0
Test suite doctest-chapter2: PASS
Test suite logged to:
/Users/xxx/code/learn4haskell/dist-newstyle/build/x86_64-osx/ghc-8.10.1/learn4haskell-0.0.0.0/t/doctest-chapter2/test/learn4haskell-0.0.0.0-doctest-chapter2.log
1 of 1 test suites (1 of 1 test cases) passed.
cabal run learn4haskell-test --enable-tests -- -m "Chapter2"
Build profile: -w ghc-8.10.1 -O1
In order, the following will be built (use -v for more details):
 - happy-1.20.0 (exe:happy) (requires build)
 - pretty-show-1.10 (lib) (requires build)
 - hedgehog-1.0.4 (lib) (requires build)
 - hspec-hedgehog-0.0.1.2 (lib) (requires build)
 - learn4haskell-0.0.0.0 (test:learn4haskell-test) (first run)
Starting     happy-1.20.0 (exe:happy)
Building     happy-1.20.0 (exe:happy)

Failed to build exe:happy from happy-1.20.0.
Build log ( /Users/xxx/.cabal/logs/ghc-8.10.1/hppy-1.20.0-9ac48e58.log ):
Configuring executable 'happy' for happy-1.20.0..
Preprocessing executable 'happy' for happy-1.20.0..
Building executable 'happy' for happy-1.20.0..
[ 1 of 19] Compiling AbsSyn           ( src/AbsSyn.lhs, dist/build/happy/happy-tmp/AbsSyn.o )
[ 2 of 19] Compiling GenUtils         ( src/GenUtils.lhs, dist/build/happy/happy-tmp/GenUtils.o )
[ 3 of 19] Compiling NameSet          ( src/NameSet.hs, dist/build/happy/happy-tmp/NameSet.o )
[ 4 of 19] Compiling ParamRules       ( src/ParamRules.hs, dist/build/happy/happy-tmp/ParamRules.o )
[ 5 of 19] Compiling ParseMonad       ( src/ParseMonad.hs, dist/build/happy/happy-tmp/ParseMonad.o )
[ 6 of 19] Compiling Lexer            ( src/Lexer.lhs, dist/build/happy/happy-tmp/Lexer.o )
[ 7 of 19] Compiling AttrGrammar      ( src/AttrGrammar.lhs, dist/build/happy/happy-tmp/AttrGrammar.o )
[ 8 of 19] Compiling AttrGrammarParser ( src/AttrGrammarParser.hs, dist/build/happy/happy-tmp/AttrGrammarParser.o )
[ 9 of 19] Compiling Grammar          ( src/Grammar.lhs, dist/build/happy/happy-tmp/Grammar.o )
[10 of 19] Compiling LALR             ( src/LALR.lhs, dist/build/happy/happy-tmp/LALR.o )
[11 of 19] Compiling First            ( src/First.lhs, dist/build/happy/happy-tmp/First.o )
[12 of 19] Compiling Parser           ( src/Parser.hs, dist/build/happy/happy-tmp/Parser.o )
[13 of 19] Compiling Paths_happy      ( dist/build/happy/autogen/Paths_happy.hs, dist/build/happy/happy-tmp/Paths_happy.o )
[14 of 19] Compiling Info             ( src/Info.lhs, dist/build/happy/happy-tmp/Info.o )
[15 of 19] Compiling PrettyGrammar    ( src/PrettyGrammar.hs, dist/build/happy/happy-tmp/PrettyGrammar.o )
[16 of 19] Compiling ProduceGLRCode   ( src/ProduceGLRCode.lhs, dist/build/happy/happy-tmp/ProduceGLRCode.o )
[17 of 19] Compiling Target           ( src/Target.lhs, dist/build/happy/happy-tmp/Target.o )
[18 of 19] Compiling ProduceCode      ( src/ProduceCode.lhs, dist/build/happy/happy-tmp/ProduceCode.o )
[19 of 19] Compiling Main             ( src/Main.lhs, dist/build/happy/happy-tmp/Main.o )
Linking dist/build/happy/happy ...
Undefined symbols for architecture x86_64:
  "___darwin_check_fd_set_overflow", referenced from:
      _awaitEvent in libHSrts.a(Select.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
`clang' failed in phase `Linker'. (Exit code: 1)
cabal: Failed to build exe:happy from happy-1.20.0 (which is required by
test:learn4haskell-test from learn4haskell-0.0.0.0). See the build log above
for details.

make: *** [test-chapter2] Error 1
```

**Comment by capital-G** [comment]
Using

```
ghcup install ghc 8.10.2
ghcup set ghc 8.10.2
ghcup install cabal 3.2.0.0
```

from README it actually works but those are not the versions distributed by brew (which I used before).

**Comment by chshersh** [comment]
@capital-G Thanks for reporting the problem! As I also see, this is not a Learn4Haskell-specific issue. Minor versions of GHC can contain some bugs, and we tested the project with GHC-8.10.2 only, that's why we recommend the version we know to be working. I see that the issue is resolved for you when using `ghc-8.10.2`, so I'm closing this issue as resolved 👌🏻
